scene/gui/file_dialog.cpp

```diff
@@ -605,40 +587,41 @@ void FileDialog::_go_forward() {
 
 void FileDialog::deselect_all() {
 	// Clear currently selected items in file manager.
-	tree->deselect_all();
+	file_list->deselect_all();
 
 	// And change get_ok title.
-	if (!tree->is_anything_selected()) {
-		get_ok_button()->set_disabled(_is_open_should_be_disabled());
+	get_ok_button()->set_disabled(_is_open_should_be_disabled());
 
-		switch (mode) {
-			case FILE_MODE_OPEN_FILE:
-			case FILE_MODE_OPEN_FILES:
-				set_internal_ok_text(ETR("Open"));
-				break;
-			case FILE_MODE_OPEN_DIR:
-				set_internal_ok_text(ETR("Select Current Folder"));
-				break;
-			case FILE_MODE_OPEN_ANY:
-				set_ok_button_text(ETR("Open"));
-				break;
-			case FILE_MODE_SAVE_FILE:
-				set_ok_button_text(ETR("Save"));
-				break;
-		}
+	switch (mode) {
+		case FILE_MODE_OPEN_FILE:
+		case FILE_MODE_OPEN_FILES:
+			set_internal_ok_text(ETR("Open"));
+			break;
+		case FILE_MODE_OPEN_DIR:
+			set_internal_ok_text(ETR("Select Current Folder"));
+			break;
+		case FILE_MODE_OPEN_ANY:
+			set_ok_button_text(ETR("Open"));
+			break;
+		case FILE_MODE_SAVE_FILE:
+			set_ok_button_text(ETR("Save"));
+			break;
 	}
 }
 
-void FileDialog::_tree_multi_selected(Object *p_object, int p_cell, bool p_selected) {
-	_tree_selected();
+int FileDialog::_get_selected_file_idx() {
+	const PackedInt32Array selected = file_list->get_selected_items();
+	return selected.is_empty() ? -1 : selected[0];
 }
 
-void FileDialog::_tree_selected() {
-	TreeItem *ti = tree->get_selected();
-	if (!ti) {
-		return;
+void FileDialog::_file_list_multi_selected(int p_item, bool p_selected) {
+	if (p_selected) {
+		_file_list_selected(p_item);
 	}
```

Comment:
This needs to update `get_ok_button()->set_disabled(_is_open_should_be_disabled());` when deselecting as well.